### PR TITLE
Add basic codemirror

### PR DESCRIPTION
**Pull request by mrdrogdrog**
This PR adds the basic CodeMirror editor and enables every plugin that is shipped with codemirror and can be enabled with typescript.  Many things aren't working right now, but the theme, the editor itself and the sizing should work.

### CODE REVIEW

File: src/components/editor/editor-window/editor-window.scss [hunk 1]
```diff
@@ -0,0 +1,6 @@
+@import '../../../../node_modules/codemirror/lib/codemirror.css';
+@import '../../../../node_modules/codemirror-one-dark-theme/one-dark.css';
+
+.CodeMirror {
+    height: 100%;
+}
```
Comment (SuperSandro2000) [inline src/components/editor/editor-window/editor-window.scss:6]:
```suggestion
}

```

Comment (ErikMichelson) [reply src/components/editor/editor-window/editor-window.scss:6]: @SuperSandro2000  You mean the newline at the EOF?

Comment (SuperSandro2000) [reply src/components/editor/editor-window/editor-window.scss:6]: Yes. I thought I did not send the comment.

File: src/components/editor/editor.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React from 'react'
 import { useSelector } from 'react-redux'
-import { useParams } from 'react-router'
+//import { useParams } from 'react-router'
```
Comment (SISheogorath) [inline src/components/editor/editor.tsx:3]: When it's commented out, it should be removed, not left in here.

File: src/components/editor/editor.tsx [hunk 3]
```diff
@@ -14,12 +14,11 @@ interface RouteParameters {
 
 const Editor: React.FC = () => {
   const editorMode: EditorMode = useSelector((state: ApplicationState) => state.editorConfig.editorMode)
-  const { id } = useParams<RouteParameters>()
+  //const { id } = useParams<RouteParameters>()
```
Comment (SISheogorath) [inline src/components/editor/editor.tsx:17]: When it's commented out, it should be removed, not left in here.

File: src/components/editor/editor-window/editor-window.tsx [hunk 4]
```diff
@@ -1,10 +1,66 @@
...
+          'CodeMirror-linenumbers',
+          'authorship-gutters',
+          'CodeMirror-foldgutter'
+        ],
+        // extraKeys: this.defaultExtraKeys,
+        flattenSpans: true,
+        addModeClass: true,
+        // autoRefresh: true,
+        // otherCursors: true
+        placeholder: "← Start by entering a title here\n===\nVisit /features if you don't know what to do.\nHappy hacking :)"
```
Comment (ErikMichelson) [inline src/components/editor/editor-window/editor-window.tsx:54]: Should we make this translateable?

Comment (DerMolly) [reply src/components/editor/editor-window/editor-window.tsx:54]: and configurable

Comment (mrdrogdrog) [reply src/components/editor/editor-window/editor-window.tsx:54]: Well yes. I can add that now. But the purpose of this PR is to add the codemirror itself and then use more PRs to form it.

Comment (DerMolly) [reply src/components/editor/editor-window/editor-window.tsx:54]: Then we should transform this to an issue and merge

**Pull request merged**